seed/energy/tasks.py

```diff
@@ -0,0 +1,237 @@
+import calendar
+import logging
+import time
+from datetime import date, timedelta, datetime
+
+from billiard import current_process
+from celery import current_app
+from celery import shared_task
+from django.conf import settings
+from django.core.cache import cache
+
+from seed.energy.meter_data_processor import green_button_data_analyser as analyser
+from seed.energy.meter_data_processor import green_button_driver as driver
+from seed.models import (
+    GreenButtonBatchRequestsInfo,
+)
+
+_log = logging.getLogger(__name__)
+
+LOCK_EXPIRE = 60 * 60 * 24 * 30  # Lock expires in 30 days
+
+
+def datetime_to_timestamp(dt):
+    return (dt - datetime.datetime(1970, 1, 1)).total_seconds()
+
+
+def increment_day(date_str):
+    """
+    Format of date_str is MM/DD/YYYY
+    """
+
+    if date_str == '' or date_str is None:
+        newdate = date.today() - timedelta(1)
+    else:
+        t = time.strptime(date_str, '%m/%d/%Y')
+        newdate = date(t.tm_year, t.tm_mon, t.tm_mday) + timedelta(1)
+
+    return newdate.strftime('%m/%d/%Y')
+
+
+@shared_task
+def green_button_task_runner():
+    # Get total number of processes and current process index, to set offset
+    # Tasks are distributed to all workers in Round-Robin style
+    _log.debug("running green_button_task_runner")
+    stats = current_app.control.inspect().stats()
+    num_process = len(stats[stats.keys()[0]]['pool']['processes'])
+    offset = current_process().index
+
+    record = ts_parser_record.objects.filter(active='Y')
```

Comment:
Hi Nicholas,

Actually the seed/energy/tasks.py is obsolete code, the latest file is
seed/energy/meter_data_processor/green_button_task.py, I found it in pull
request #678...

The ts_parser_record is renamed to GreenButtonBatchRequestsInfo, it's a
SEED model defined at the bottom of seed/model.py. The according table in
PostgreSQL holds the records needed by parser to request GreenButton data.

Best,
Haopeng

On Thu, Feb 11, 2016 at 11:23 AM, Nicholas Long <notifications@github.com>
wrote:

> In seed/energy/tasks.py
> <https://github.com/SEED-platform/seed/pull/714#discussion_r52625134>:
>
> > +        t = time.strptime(date_str, '%m/%d/%Y')
> > +        newdate = date(t.tm_year, t.tm_mon, t.tm_mday) + timedelta(1)
> > +
> > +    return newdate.strftime('%m/%d/%Y')
> > +
> > +
> > +@shared_task
> > +def green_button_task_runner():
> > +    # Get total number of processes and current process index, to set offset
> > +    # Tasks are distributed to all workers in Round-Robin style
> > +    _log.debug("running green_button_task_runner")
> > +    stats = current_app.control.inspect().stats()
> > +    num_process = len(stats[stats.keys()[0]]['pool']['processes'])
> > +    offset = current_process().index
> > +
> > +    record = ts_parser_record.objects.filter(active='Y')
>
> @wanghp18 <https://github.com/wanghp18>: That makes sense. I think the
> issue is that the ts_parser_record does not exist. Can you grep your code
> for where you define that class/model?
>
> Let me know if you would like to chat today--it may be faster
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/SEED-platform/seed/pull/714/files#r52625134>.
>